lib/includes/json-helper.js

```diff
@@ -194,20 +194,20 @@ export const FHIRHelper = {
 
   /**
    * Creates an ES6 class instance based on a value extracted from the JSON.
-   * @param {string} key - the original key under which the value was stored.  This is used as a backup in case the value
-   *   does not declare its type.
-   * @param {object} value - the FHIR data to create an ES6 class instance for
+   * @param {string} shrType - the fqn of the class to be instantiated
+   * @param {object} fhir - the FHIR data to create an ES6 class instance for
+   * @param {string} fhirType - the type of the FHIR object passed in, just in case it's not otherwise available by inspecting the object
    * @returns {object} an instance of an ES6 class representing the data
    * @private
    */
-  createInstanceFromFHIR: function(key, value, shrId, allEntries=[], mappedResources={}, referencesOut=[], asExtension=false) {
-    if (Array.isArray(value)) {
-      return value.map(v => FHIRHelper.createInstanceFromFHIR(key, v, shrId, allEntries, mappedResources, referencesOut, asExtension));
+  createInstanceFromFHIR: function(shrType, fhir, fhirType, shrId, allEntries=[], mappedResources={}, referencesOut=[], asExtension=false) {
```

Comment:
The code has three basic implementations of functions called `createInstanceFromFHIR`: this one, a high-level one generated on `ObjectFactory`, and one on each `NamespaceFactory`.

The latter two have signature:
```
function(fhir, fhirType, shrType, shrId=uuid(), allEntries=[], mappedResources={}, referencesOut=[], asExtension=false)
```
But this one has signature:
```
function(shrType, fhir, fhirType, shrId, allEntries=[], mappedResources={}, referencesOut=[], asExtension=false)
```

Note that they are almost exactly the same except for the order of the first three arguments.  Ideally, there should be consistency between all of them in order to avoid stupid and unintentional mistakes (probably to be made by me).

So... can we either change this signature to match the others or change the others to match this one?

Reply:
I updated the other locations to match this one, I think it's fewer code changes and in simpler places, plus the ordering feels more consistent with how `createInstanceFromFHIR('ns.Class', fhir, ...)` turns into `Class.fromFHIR(fhir, ....)` .   
Also I updated the comments for `shrType` similarly to the above, originally they said "optional" but it can be required depending on other parameters.